controllers/namespacelabel_controller.go

```diff
@@ -65,81 +64,101 @@ type NamespaceLabelReconciler struct {
 // This reconcile function adds the labels from the NamespaceLabel to the namespace it runs against,
 // and deletes the labels when the resource is deleted.
 func (r *NamespaceLabelReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := logrus.New()
-	log.SetFormatter(&ecslogrus.Formatter{})
-
+	log := ctrllog.FromContext(ctx)
 	var namespaceLabel danaiodanaiov1alpha1.NamespaceLabel
 	if err := r.Get(ctx, req.NamespacedName, &namespaceLabel); err != nil {
 		if k8serrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		log.WithError(errors.New(err.Error())).WithFields(logrus.Fields{"NamespaceLabel": req.NamespacedName}).Error("unable to fetch NamespaceLabel")
+		log.Error(err, "Unable to fetch NamespaceLabel", "namespaceLabel", req.NamespacedName.String())
 		return ctrl.Result{Requeue: true}, client.IgnoreNotFound(err)
 	}
 
 	var namespace corev1.Namespace
 	if err := r.Client.Get(ctx, types.NamespacedName{Name: namespaceLabel.ObjectMeta.Namespace}, &namespace); err != nil {
-		log.WithError(errors.New(err.Error())).WithFields(logrus.Fields{"namespace": namespaceLabel.ObjectMeta.Namespace}).Error("unable to fetch namespace while getting previous labels")
+		log.Error(err, "Unable to fetch namespace", "namespace", namespaceLabel.ObjectMeta.Namespace)
 		return ctrl.Result{Requeue: true}, client.IgnoreNotFound(err)
 	}
+	if namespaceLabel.ObjectMeta.DeletionTimestamp.IsZero() {
+		if !controllerutil.ContainsFinalizer(&namespaceLabel, FinalizerName) {
+			if err := r.addFinalizer(ctx, log, &namespaceLabel, &namespace); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		if controllerutil.ContainsFinalizer(&namespaceLabel, FinalizerName) {
+			if res, err := r.deleteLabels(ctx, log, &namespaceLabel, &namespace); err != nil {
+				return res, err
+			}
+			return ctrl.Result{}, nil
+		}
+	}
```

Comment:
I think you can switch the condition to check if it _is_ being deleted and since you return in that case, you don't need an if-else and you can just do

```
if !namespaceLabel.ObjectMeta.DeletionTimestamp.IsZero() {
	if controllerutil.ContainsFinalizer(&namespaceLabel, FinalizerName) {
		if res, err := r.deleteLabels(ctx, log, &namespaceLabel, &namespace); err != nil {
			return res, err
		}
		return ctrl.Result{}, nil
	}

if !controllerutil.ContainsFinalizer(&namespaceLabel, FinalizerName) {
	if err := r.addFinalizer(ctx, log, &namespaceLabel, &namespace); err != nil {
		return ctrl.Result{}, err
	}
}
```

Reply:
updated in latest pr